src/Core/CFamily/EffectiveRulesConfigCalculator.cs

```diff
@@ -115,7 +116,7 @@ public CacheEntry(ICFamilyRulesConfig sourceConfig, RulesSettings sourceSettings
                 public ICFamilyRulesConfig EffectiveConfig { get; }
             }
 
-            private readonly IDictionary<string, CacheEntry> languageToConfigMap = new Dictionary<string, CacheEntry>();
+            private readonly IDictionary<string, CacheEntry> languageToConfigMap = new ConcurrentDictionary<string, CacheEntry>();
```

Comment:
I think it's worth adding a note in the method to explain the potential race conditions and consequences  - perhaps something like:

> GetEffectiveRulesConfig could be called simultaneously from multiple threads. This won't cause a crash because we are using a ConcurrentDictionary (see bug #2783).  It could mean that we are doing unnecessary work with multiple threads calculating the same result.